src/layers/core/grid-layer/grid-layer.js

```diff
@@ -54,25 +56,62 @@ function _needsReProjectPoints(oldProps, props) {
   return oldProps.cellSize !== props.cellSize;
 }
 
+function _percentileChanged(oldProps, props) {
+  return oldProps.lowerPercentile !== props.lowerPercentile ||
+    oldProps.upperPercentile !== props.upperPercentile;
+}
+
+function _needsReSortBins(oldProps, props) {
+  return oldProps.getColorValue !== props.getColorValue;
```

Comment:
> we have no way to detect whether some one pass in a new function to calculate color value.

Unless we expand `updateTriggers` to cover this case (but that would probably require a discussion and a separate PR).

OK for now, but I think it is a potential API design issue that we have different behavior for accessors and other functions. We might want to call it out in one of our documents (possibly API guidelines if you decided to update there).